### ISSUE TITLE
Order of atomicfu and kotlin plugins in the classpath 

### ISSUE THREAD

**Issue by mvicsokolova** [Kotlin/kotlinx-atomicfu#384]
After #377 a strange behaviour reproduced during application of `atomicfu-gradle-plugin` to kotlinx.coroutines. 
The order in which `atomicfu-gradle-plugin` and `kotlin-gradle-plugin` are added to the classpath defines actual dependencies found on the classpath.
I've described the problem in detail here: https://github.com/Kotlin/kotlinx.coroutines/pull/3984.

The actual reason of this behaviour should be investigated.

**Comment by mvicsokolova** [comment]
The high level reason of why this problem occurred is that the compiler plugin has the same id as the `atomicfu-gradle-plugin` (See the name of the [file](https://github.com/JetBrains/kotlin/blob/236a0ca6f0f30cdf1eac5d117ecb83d6de5e4f68/libraries/tools/atomicfu/src/common/resources/META-INF/gradle-plugins/kotlinx-atomicfu.properties#L1) that provides the implementation class for the compiler plugin). 🙃 Hence this id clash causes a conflict during plugin resolution and the actual atomicfu-gradle-plugin is not applied at all.

I can suggest the temporary solution for this issue: 
we just write a big message in README: "Please tweak the plugin order in case of the compilation error (or do not apply kotlin-gradle-plugin in the legacy manner via the build script)". And the good solution with migration will follow (see this [comment](https://github.com/Kotlin/kotlinx-atomicfu/issues/370#issuecomment-1862776597)).


Or the alternative solution will be to create **AtomicfuGradlePluginLegacy** with the different id, e.g. "kotlinx-atomicfu-legacy". And warn users in case of the compilation error to use the plugin with the id `kotlinx-atomicfu-legacy`.

**Pros:**
- the conflict of ids will be avoided
**Cons:**
- I guess, when you suggest users to use something deprecated or legacy, you need to provide a preferable option to use, though this version (in the form of the compiler plugin) is not ready yet.

Both solutions are bad, and in this case I'd choose the first one, because it's easier.

**Comment by mainrs** [comment]
I want to chime in and share a bug I came across when applying the plugin as described in the readme file.

I added it to `buildscript.dependencies.classpath` and I then went ahead and applied the plugin to all projects using `allprojects.apply(plugin = "kotlinx-atomicfu")`. This broke compatibility with Java source. One of the modules of my project contained a single java class file under `src/main/java` and that file was not part of the classpath anymore. The project compiled file, but running the program resulted in a `NoClassDefFoundError`. And the jar I build using the `build` command did not contain said java class file.

Maybe this is a misconfiguration on my side...
